java/src/jmri/jmrit/dispatcher/AutoActiveTrain.java

```diff
@@ -1410,9 +1410,12 @@ public void run() {
                 }
                 executeStopTasks(_task);
             } catch (InterruptedException e) {
-                // interrupting will cause termination without executing the task
+                log.warn("Waiting for train to stop interupted - stop tasks not executing");
+            } catch (Exception e) {
+                log.error("Waiting for train to stop crashed - stop tasks not executing [{}] ", e.getMessage());
```

Comment:
If we get to this situation, we probably want a complete stack trace. I recommend replacing the error with:
```
log.error("Waiting for train to stop crashed - stop tasks not executing.", e);
```

Reply:
Thread.dumpStack()

Reply:
999 to -1 or Integer.MAX_VALUE.

Reply:
@geowar1 We don't want to use Thread.dumpStack() because that goes to Standard Error, not to the logs.

Reply:
IIRC there's a way to send it to the logs… 
new Exception("Stack trace").printStackTrace(System.out);

(from <https://stackoverflow.com/questions/54882/how-do-i-in-java-add-a-stacktrace-to-my-debugging-printout>)

Humm… never mind; apparently just logging the exception will dump the stack also (I'm guessing you already knew that… ;-):
log.error("error here", e); // print the stack trace to your log.

Reply:
Right. Will do the log.error("error here", e); // print the stack trace to your log. type.
I would prefer to use integer.MAX_VALUE than -1 which generally means the field is in error, but maybe thats habit. In reality the limit is 150, which is the maximum number of sections in a transit, I didn't set that but I noticed it in the code.
While I have you attention, I used collections.sort for sorting a list, but I notice its not the most popular sort solution. Is there a preferred method? And, in the beantable area there is rbx.getString() and bundle.getMessage, which one should I use for what as I need to add a filed label there.
Thanks.
ps I wont push anything till tomorrow morning unless by accident.

Reply:
Quick google search said collections.sort… but it's probably dependent on the data that you're sorting. If it's just Strings and fewer than a thousand or so I'd just use collections.sort().
IIRC we're trying to move away from rb[x].getString and always go with bundle.getMessage().

Reply:
I suggested -1 because another later special condition could be -2 and the -3... (I don't know that there would ever be additional special conditions, but that is how I tend to think.)

New in Java 1.8 is [```List.sort(Compator)```](https://docs.oracle.com/javase/8/docs/api/java/util/List.html#sort-java.util.Comparator-). I would use that going forward, and (mostly) ignore our prior art which predates this method.

Reply:
Well, I'll give the list.sort a go, and think about the -1 and see if that works.